internal/backends/winit/winitwindowadapter.rs

```diff
@@ -1134,13 +1190,19 @@ impl WindowAdapterInternal for WinitWindowAdapter {
 
     #[cfg(muda)]
     fn setup_menubar(&self, menubar: vtable::VBox<i_slint_core::menus::MenuVTable>) {
-        drop(self.muda_adapter.borrow_mut().take());
-        self.muda_adapter.replace(Some(crate::muda::MudaAdapter::setup(
-            menubar,
-            &self.winit_window().unwrap(),
-            self.event_loop_proxy.clone(),
-            self.self_weak.clone(),
-        )));
+        drop(self.menubar.borrow_mut().take());
+        self.menubar.replace(Some(menubar));
+
+        if let WinitWindowOrNone::HasWindow { muda_adapter, .. } =
+            &*self.winit_window_or_none.borrow()
+        {
+            muda_adapter.replace(Some(crate::muda::MudaAdapter::setup(
```

Comment:
But here you might need to delete the muda adapter first


```suggestion
            // On Windows, we must destroy the muda menu before re-creating a new one
            drop(self.muda_adapter.borrow_mut().take());
            muda_adapter.replace(Some(crate::muda::MudaAdapter::setup(
```